src/event/writer.rs

```diff
@@ -161,9 +172,39 @@ impl EventWriter {
                 .expect("send error");
             rx_error
         } else {
+            self.event_handles.push_back(rx_flush);
             rx
         }
     }
+
+    /// Flush data.
+    ///
+    /// It will wait until all pending appends have acknowledgment.
+    ///
+    /// # Examples
+    /// ```ignore
+    /// let mut byte_writer = client_factory.create_byte_writer(segment).await;
+    /// let payload = vec![0; 8];
+    /// let size = event_writer.write_event(&payload).await;
+    /// event_writer.flush().await;
+    /// ```
+    pub async fn flush(&mut self) -> Result<(), Error> {
+        while self.event_handles.front().is_some() {
+            let mut receiver = self.event_handles.swap_remove_front(0).expect("get first handle");
```

Comment:
I think `pop_front()` would be preferable to `swap_remove_front(0)` here.
This could also be combined with the conditional as in:
```
while let Some(receiver) = self.event_handles.front()
```
To be more efficient, clear, and concise.